library/account/src/main/kotlin/org/cru/godtools/account/GodToolsAccountManager.kt

```diff
@@ -75,12 +85,20 @@ class GodToolsAccountManager @VisibleForTesting internal constructor(
     ): ActivityResultLauncher<AccountType> {
         val launchers = providers.associate {
             it.type to it.rememberLauncherForLogin(createAccount) { result ->
+                val exceptionOrNull = result.exceptionOrNull()
                 onResponse(
                     when {
                         result.isSuccess -> LoginResponse.Success
-                        else -> when (result.exceptionOrNull()) {
-                            AuthenticationException.UserNotFound -> LoginResponse.Error.UserNotFound
-                            AuthenticationException.UserAlreadyExists -> LoginResponse.Error.UserAlreadyExists
+                        else -> when {
+                            exceptionOrNull == AuthenticationException.UserNotFound -> {
+                                LoginResponse.Error.UserNotFound
+                            }
+
+                            exceptionOrNull == AuthenticationException.UserAlreadyExists -> {
+                                LoginResponse.Error.UserAlreadyExists
+                            }
+
+                            !isConnectedFlow.value -> LoginResponse.Error.NotConnected
                             else -> LoginResponse.Error()
```

Comment:
```suggestion
                        else -> when (result.exceptionOrNull()) {
                            AuthenticationException.UserNotFound -> LoginResponse.Error.UserNotFound
                            AuthenticationException.UserAlreadyExists -> LoginResponse.Error.UserAlreadyExists
                            !isConnectedFlow.value -> LoginResponse.Error.NotConnected
                            else -> when {
                                isConnectedFlow.value -> LoginResponse.Error.NotConnected
                                else -> LoginResponse.Error()
                            }
```

Reply:
I'm actually not a fan of using a `StateFlow` like this, by collecting eagerly when the AccountManager is created you will always be monitoring the online status even if the user never touches the login screen.

There is no need to continuously monitor the online state, it should just be a single check to see if the user is online when an error is encountered